crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -291,19 +295,62 @@ mod orderbook {
             };
         }
 
-        fn fullfil_order(
-            ref self: ContractState,
-            order_hash: felt252,
-            execution_info: ExecutionInfo,
-            signer: Signer
-        ) {}
+        fn fulfill_order(ref self: ContractState, fulfill_info: FulfillInfo, signer: Signer) {
+            let order_hash = fulfill_info.order_hash;
+            let execution_hash = fulfill_info.hash();
+            SignerValidator::verify(execution_hash, signer);
+            let order: OrderV1 = match order_read(order_hash) {
+                Option::Some(o) => o,
+                Option::None => panic_with_felt252(orderbook_errors::ORDER_NOT_FOUND),
+            };
+            let status = match order_status_read(order_hash) {
+                Option::Some(s) => s,
+                Option::None => panic_with_felt252(orderbook_errors::ORDER_NOT_FOUND),
+            };
+            assert(status == OrderStatus::Open, orderbook_errors::ORDER_NOT_FULFILLABLE);
+            let order_type = match order_type_read(order_hash) {
+                Option::Some(s) => s,
+                Option::None => panic_with_felt252(orderbook_errors::ORDER_NOT_FOUND),
+            };
+            let token_hash = order.compute_token_hash();
```

Comment:
`token_hash` seems not to be used and `order` is only used to compute the `token_hash`

Reply:
true